tao_compiler/mlir/xla/ral/context/stream_executor_based_impl.cc

```diff
@@ -401,6 +404,64 @@ void ral_gemm(ExecutionContext* ctx, void* stream_handle, MemRefType<InT, 2> A,
   }
 }
 
+void ral_qgemm(
+    ExecutionContext* ctx, void* stream_handle, MemRefType<int8_t, 2> input,
+    MemRefType<int8_t, 2> weight, MemRefType<float, 0> inputScales,
+    MemRefType<int32_t, 0> inputZeroPoints, MemRefType<float, 1> weightScales,
+    MemRefType<int32_t, 1> weightZeroPoints, MemRefType<float, 0> resultScales,
+    MemRefType<int32_t, 0> resultZeroPoints, MemRefType<int8_t, 2> result,
+    bool tp_a, bool tp_b, bool weight_is_const) {
+  if (isEmptyMemref(input) || isEmptyMemref(weight) || isEmptyMemref(result)) {
+    TAO_VLOG(1) << "ral_qgemm: early return for empty tensor";
+    return;
+  }
+  TAO_VLOG(0) << "tp_a: " << tp_a << " tp_b: " << tp_b;
+  TAO_VLOG(0) << "tp_a: " << tp_a << " tp_b: " << tp_b;
+  TAO_VLOG(0) << "input.sizes[0]: " << input.sizes[0]
+              << " input.sizes[1]: " << input.sizes[1];
+  TAO_VLOG(0) << "weight.sizes[0]: " << weight.sizes[0]
+              << " weight.sizes[1]: " << weight.sizes[1];
+  TAO_VLOG(0) << "result.sizes[0]: " << result.sizes[0]
+              << " result.sizes[1]: " << result.sizes[1];
+
+  auto gpu_driver = ctx->getDriver<GPUDriver>(GPUDriver::name());
+  float input_scale, weight_scale, result_scale;
+  gpu_driver->d2h(ctx, stream_handle, &inputScales.data[0], &input_scale,
+                  sizeof(float));
+  gpu_driver->d2h(ctx, stream_handle, &weightScales.data[0], &weight_scale,
```

Comment:
I see the UT using per-tensor quantization  for weight while we only handle the first element of the scale. Is it the supposed behavior?

Reply:
Yes, we only support per tensor quantization on gpu. This ut is used for per tensor testing although its weight scale is per channel.

Reply:
Then why not use a per-tensor UT directly？the compiler side is supposed to be ok for both cases. And the register kernel should use `memref<f32>` for weight scale/zero_point inputs.

Reply:
Of course we can use a per-tensor UT directly. However, I think per-channel/per-tensor should be unified in library call interface although we may launch different kernels for them. And if we support per-channel quantization later, we only need to modify ral kernel library call function.